cmd/debugger/main.go

```diff
@@ -1,58 +1,129 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"net"
 	"os"
 	"os/exec"
+	"time"
+
+	"github.com/earthly/earthly/debugger/common"
+
+	"github.com/creack/pty"
+	"github.com/hashicorp/yamux"
 )
 
 var (
 	// Version is the version of the debugger
 	Version string
+
+	// ErrNoShellFound occurs when the container has no shell
+	ErrNoShellFound = fmt.Errorf("no shell found")
 )
 
-func interactiveMode(remoteConsoleAddr string) {
-	c, err := net.Dial("tcp", remoteConsoleAddr)
+func getShellPath() (string, bool) {
+	for _, sh := range []string{
+		"bash", "ksh", "zsh", "sh",
+	} {
+		if path, err := exec.LookPath(sh); err == nil {
+			return path, true
+		}
+	}
+	return "", false
+}
+
+func interactiveMode(remoteConsoleAddr string) error {
+	conn, err := net.Dial("tcp", remoteConsoleAddr)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to connect to earth remote console: %v\n", err)
-		return
+		return err
```

Comment:
It might be a good idea to `errors.Wrap` all errors coming from other packages. We've done this in all other places in our code. It helps with tracking issues.

In this case, we should additionally specify the address we were trying to connect to.

Reply:
good point, i'll try to remember this in the future.